core/Pos/Core/Common/Types.hs

```diff
@@ -206,6 +351,17 @@ instance NFData AddrAttributes
 instance NFData AddrStakeDistribution
 instance NFData Address
 
+instance Bi Address where
+    encode Address{..} =
+        Bi.encodeCrcProtected (addrRoot, addrAttributes, addrType)
+    decode = do
+        (addrRoot, addrAttributes, addrType) <- Bi.decodeCrcProtected
+        let res = Address {..}
+        pure res
```

Comment:
Would that not be simpler (and remove the `RecordWildCard` usage that I dislike) to write that as:
```
    decode = do
        (aRoot, attrs, aType) <- Bi.decodeCrcProtected
        pure $ Address aRoot attrs aType
```
its the same level of type safety, but i find the second more readable. I do realise that you are probably just moving this code from another module to this one, but it does provide an opportunity to make minor fixes like this.